x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingResourceTracker.java

```diff
@@ -243,11 +255,39 @@ static void getMemoryAndProcessors(
                 if (Priority.LOW.equals(modelAssignment.getValue().getTaskParams().getPriority()) == false) {
                     // as assignments can be placed on different nodes, we only need numberOfThreadsPerAllocation here
                     extraSingleNodeProcessors = Math.max(extraSingleNodeProcessors, numberOfThreadsPerAllocation);
-                    extraProcessors += numberOfAllocations * numberOfThreadsPerAllocation;
+                    extraProcessors += numberOfRequestedAllocations * numberOfThreadsPerAllocation;
                 }
             } else if (assignment.getNodeRoutingTable().values().stream().allMatch(r -> r.getState().consumesMemory() == false)) {
                 // Ignore states that don't consume memory, for example all allocations are failed
                 continue;
+            } else if (AssignmentState.STARTED.equals(assignment.getAssignmentState()) && numMissingAllocations > 0) {
```

Comment:
I decided to add logging in this case at the warn and info levels because this case only occurs while we are actively scaling up. If the scaling doesn't work, we will want to see these logs. The other logging should remain at debug level because this function gets called every 5 seconds and would cause a lot of logs.